ballerina/listener.bal

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2025 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+import ballerina/uuid;
+
+# Listener for task scheduling.
+public class Listener {
+    private ListenerConfiguration config;
```

Comment:
Unless we are re-using this in any of the listener methods (other than the init) I think we could hide this as native data and remove this field.